README.md

```diff
@@ -1,16 +1,688 @@
+  v   SemVer  -- Version
+  run Scope   -- Which actions to invoke
+  nnc String  -- Nonce
+  ext nullable Any (implicit null) -- Extended fields
+}
+
+type Scope enum {
+  | All ("*")
+  | Action
+}
+
+type Action enum {
+  with  URI 
+  do     Ability
+  inputs Any
+} 
+```
+
+## 3.3 JSON Examples
+
+### 3.3.1 Run All
+
+ ``` js
+{
+  "ucan/invoke": {
+    "v": "0.1.0",
+    "prf": [{"/": "bafkreifzjut3te2nhyekklss27nh3k72ysco7y32koao5eei66wof36n5e"}],
+    "nnc": "abcdef",
+    "run": "*", // Explicitly "run all"
+    "ext": null
+  },
+  "sig": { "/": { "bytes:": "bdNVZn_uTrQ8bgq5LocO2y3gqIyuEtvYWRUH9YT-SRK6v_SX8bjt-VZ9JIPVTdxkWb6nhVKBt6JGpgnjABpOCA" } }
+}
+```
+
+### 3.3.2 Promise Pipelines
+
+Promise pipelines are handled in more detail in [section 5](#5-promise-pipelining). In brief, they enable taking the output of an action that has not yet run as the input to another action. Here is a simple example:
+
+``` js
+{
+  "ucan/invoke": {
+    "v": "0.1.0",
+    "prf": [{"/": "bafkreifzjut3te2nhyekklss27nh3k72ysco7y32koao5eei66wof36n5e"}],
+    "nnc": "02468",
+    "ext": null,
+    "run": {
+      "notify-bob": {
+        "with": "mailto://alice@example.com",
+        "do": "msg/send",
+        "inputs": [
+          {
+            "to": "bob@example.com",
+            "subject": "DNSLink for example.com",
+            "body": "Hello Bob!"
+          }
+        ]
+      },
+      "log-as-done": {
+        "with": "https://example.com/report"
+        "do": "crud/update"
+        "inputs": {
+          "from": "mailto://alice@exmaple.com",
+          "to": ["bob@exmaple.com"],
+          "event": "email-notification",
+          "value": {"ucan/promise": ["/", "notify-bob"]} // Pipelined promise
+        }
+      }
+    }
+  },
+  "sig": {"/": {"bytes:": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"}}
+}
+```
+
+# 4 Receipt
+
+An invocation receipt is an attestation of the output of an invocation. A receipt MUST be signed by the executor (the `aud` of the associated UCAN).
+
+**NB: a Receipt this does not guarantee correctness of the result!** The statement's veracity MUST be only understood as an attestation from the executor.
+
+Receipts don't have their own version field. Receipts MUST use the same version as the invocation that they contain.
+
+## 4.1 Fields
+
+| Field  | Type            | Description                                                             | Required | Default |
+|--------|-----------------|-------------------------------------------------------------------------|----------|---------|
+| `inv`  | `&Invocation`   | CID of the Invocation that generated this response                      | Yes      |         |
+| `rlt`  | `{String: Any}` | The results of each call, the action's label. MAY contain sub-receipts. | Yes      |         |
+| `meta` | `Any`           | Non-normative extended fields                                           | No       | `null`  |
```

Comment:
@Gozala you had mentioned wanting a Receipt in general! I'm guessing that you won't want to wrap an Invocation around it for the DAG House use case?

Should we break this out into a separate spec, and generalize it to something like this:

```json
type Receipt struct {
  tsk  String -- Task description type (UCAN Invocation, something else)
  src  &CID -- Task description
  out  Any  -- Output
  meta Any (implicit Null) -- Arbitrary metadata
}
```

In either case `out` is a way better field name than `rlt` :wink: Changing _that_

Reply:
Also, do we need a standard way of signalling failure or that a task is incomplete but has started?

Reply:
('scuse the snipe) btw `&Any` or `Link` (which `&Any` is basically an alias for) is how to express an untyped link to something

Reply:
Good to know! Above it's not meant to be a `&Any` — just a user defined set of fields. Does that need to happen by reference in IPLD?

Reply:
no, `Any` should be fine for that I think? I was looking specifically at `&CID` in your comment

Reply:
Ooooh gotcha! Thanks :D

Reply:
> @Gozala you had mentioned wanting a Receipt in general! I'm guessing that you won't want to wrap an Invocation around it for the DAG House use case?

I'm not sure I understand the question, but `Receipt` struct is more or less what I would want. I'm assuming `src` is link to the invocation. Not what would I do with `tsk`though.

Reply:
> Also, do we need a standard way of signalling failure or that a task is incomplete but has started?

Ucanto has a standard Result type for indicating success / failure. In our case invocation is a tuple of capabilities been invoked and `out` is a mapped tuple of `Result` types for each invocation.

Switching from tuples to `Receipt`s is my next task after one I'm currently on.